### ISSUE TITLE
Image pull fails due to sha256 mismatch

### ISSUE THREAD

**Issue by mohafarooq** [dragonflyoss/Dragonfly2#784]
### Bug report:

FATAL:   Unable to handle oras://registry.com/project/repo/image:tag uri: unable to Download Image: unable to pull from registry: failed commit on ref "image.sif": unexpected commit digest sha256:a84aac14edcdf80c269bd1828a987dbac5dbee499076eb2d86f67bf9dabd0c47, expected sha256:4ab8dc5ab7ef604d2b10a1548de407c4c49e6638ed69ee18b241dc203c16868d: failed precondition

```
# sha256sum /dev/shm/sbuild-tmp-cache-988715426
a84aac14edcdf80c269bd1828a987dbac5dbee499076eb2d86f67bf9dabd0c47
```

### Environment:
- Dragonfly version: alpha.7
- OS:
- Kernel (e.g. `uname -a`):
- Others:

**Comment by mohafarooq** [comment]
SHA in Dfdaemon:
a84aac14edcdf80c269bd1828a987dbac5dbee499076eb2d86f67bf9dabd0c47  /dev/shm/dragonfly/lib/dragonfly/375fc2b79210bce0524323335c11d122af5347253c45dee62d8a6a2b090c186d/x.y.z.61-1-09198ec3-42b2-4395-af69-2a98f6fd6ac6/data

SHA in CDN:
# sha256sum /tmp/cdn/ftp/download/375/375fc2b79210bce0524323335c11d122af5347253c45dee62d8a6a2b090c186d
4ab8dc5ab7ef604d2b10a1548de407c4c49e6638ed69ee18b241dc203c16868d  /tmp/cdn/ftp/download/375/375fc2b79210bce0524323335c11d122af5347253c45dee62d8a6a2b090c186d

**Comment by mohafarooq** [comment]
Metadata file with sha issues:
[metadata_ShaIssue.txt](https://github.com/dragonflyoss/Dragonfly2/files/7501647/metadata_ShaIssue.txt)

Metadata file with NO sha issues:
[metadata_NoShaIssue.txt](https://github.com/dragonflyoss/Dragonfly2/files/7501650/metadata_NoShaIssue.txt)

**Comment by mohafarooq** [comment]
[md5.txt](https://github.com/dragonflyoss/Dragonfly2/files/7503054/md5.txt)

**Comment by mohafarooq** [comment]
md5 of a broken image
[md5.txt](https://github.com/dragonflyoss/Dragonfly2/files/7503401/md5.txt)

**Comment by mohafarooq** [comment]
[core.log.gz](https://github.com/dragonflyoss/Dragonfly2/files/7503804/core.log.gz)
[md5_brokenimage.txt](https://github.com/dragonflyoss/Dragonfly2/files/7503805/md5_brokenimage.txt)
[metadata.txt](https://github.com/dragonflyoss/Dragonfly2/files/7503806/metadata.txt)

**Comment by mohafarooq** [comment]
peer x.y.z.57
[core.log.gz](https://github.com/dragonflyoss/Dragonfly2/files/7504757/core.log.gz)

**Comment by jim3ma** [comment]
Fixed in #787
